### DISCUSSION TITLE
Support Non LTS Java versions

### DISCUSSION THREAD

**Discussion by theoparis** [neoforged/NeoForge#1754] [Ideas]
It would be great if neoforge didn't force a specific java toolchain. There are legimate use cases such as using the Java Foreign Function API, which only works in newer versions of Java.  Currently when trying to use neoforge to develop a mod like this:

```
      > Cannot find a Java installation on your machine matching this tasks requirements: {languageVersion=21, vendor=any vendor, implementation=vendor-specific} for LINUX on x86_64.
         > No locally installed toolchains match and toolchain download repositories have not been configured.
```

**Comment by theoparis** [comment]


**Reply by ChiefArug** [reply]
You should not have your mod target a different version of Java, and NeoForge does not provide support for such.
Mojang ship a version of Java with the launcher and that is the version that mods should target, as having users change the version is a very difficult task for the average user, and on some launchers (like CurseForge) doing so is not even possible.

Saying that, its possible the errors are coming from gradle tools who want specific java versions (unrelated to the version you compile with), in which case I suggest using the foojay resovler plugin, [as used in the official mdk](https://github.com/NeoForgeMDKs/MDK-1.21.10-NeoGradle/blob/33967085a752c806aea2d449aba17c17683c894d/settings.gradle#L9), to manage jdk version for gradle tools.
